picard/metadata.py

```diff
@@ -199,6 +200,27 @@ def compare_to_release(self, release, weights):
         sim = linear_combination_of_weights(parts) * get_score(release)
         return SimMatchRelease(similarity=sim, release=release)
 
+    @staticmethod
+    def extract_year_from_date(dt):
```

Comment:
You can just use dateutil parser and then get the year.

Reply:
Oh, this too ;)

Reply:
https://dateutil.readthedocs.io/en/stable/parser.html#module-dateutil.parser

I would much rather have this (which is thoroughly tested and supported) than trying to implement a bunch of regexes.

Reply:
Done, but I'm surprised we prefer to add a dependency on a third party library for such a small use.

Reply:
> Done, but I'm surprised we prefer to add a dependency on a third party library for such a small use.

It was a suggestion, and I see nothing wrong in using regexes (regexes engines are well-tested too).
But Python has tons of good modules available, and it is good practice to re-use existing stuff whenever possible, but I'm reluctant to rely on third-party libs for another reason: **can anyone there confirm no licensing conflict?**

Reply:
> **can anyone there confirm no licensing conflict?**

We can use this under the https://opensource.org/licenses/BSD-3-Clause, which we can use in our GPL code.

Reply:
I must say I'm also a bit concerned about adding another dependency just for this, so far we had managed to keep the external dependencies to a minimum for Picard. On the other hand we really should review other uses of date handling in our code, I think Picard currently relies on the assumption date is always `YYYY-MM-DD` or any shorter form of this (without day or without month and day). But this might not be true. So having a stable library for date handling might make sense. Two things though:

- `extract_year_from_date` probably should be moved to utils
- `dateutil` must be added to requirements*.txt

Reply:
@RaysDev  For the requirements we actually need to update not only the requirements.txt but also requirements-win.txt and requirements-macos.txt. We use the later ones to freeze the dependencies for those operating systems as we often encounter issues with specific versions. But for general development we use the requirements.txt as we try to go with newer versions whenever possible.